frontend/test/resolution/testDomains.cpp

```diff
@@ -154,109 +155,155 @@ module M {
   }
 
   assert(guard.realizeErrors() == 0);
+}
+
+static void testDomainLiteral(Context* context,
+                                  std::string domainLiteral,
+                                  DomainType::Kind domainKind) {
+  printf("Testing: %s\n", domainLiteral.c_str());
+
+  context->advanceToNextRevision(false);
+  setupModuleSearchPaths(context, false, false, {}, {});
+  ErrorGuard guard(context);
 
-  printf("Success: %s\n", domainType.c_str());
+  std::string program =
+R"""(
+module M {
+  var d = )""" + domainLiteral + R"""(;
+
+  type i = d.idxType;
+  param rk = d.isRectangular();
+  param ak = d.isAssociative();
 }
+)""";
 
-// static void testAssociative(Context* context,
-//                                   std::string domainType,
-//                                   std::string idxType,
-//                                   bool parSafe) {
-//   context->advanceToNextRevision(false);
-//   setupModuleSearchPaths(context, false, false, {}, {});
-//   ErrorGuard guard(context);
+  auto path = UniqueString::get(context, "input.chpl");
+  setFileText(context, path, std::move(program));
 
-//   std::string program =
-// R"""(
-// module M {
-//   var d : )""" + domainType + R"""(;
-//   type ig = )""" + idxType + R"""(;
+  const ModuleVec& vec = parseToplevel(context, path);
+  const Module* m = vec[0];
 
-//   type i = d.idxType;
-//   param s = d.parSafe;
-//   param rk = d.isRectangular();
-//   param ak = d.isAssociative();
+  const ResolutionResultByPostorderID& rr = resolveModule(context, m->id());
 
-//   var p = d.pid();
+  const Variable* d = m->stmt(0)->toVariable();
+  assert(d);
+  assert(d->name() == "d");
 
-//   for loopI in d {
-//     var z = loopI;
-//   }
+  QualifiedType dQt = rr.byAst(d).type();
+  assert(dQt.type());
+  auto dType = dQt.type()->toDomainType();
+  assert(dType);
 
-//   proc generic(arg: domain) {
-//     type GT = arg.type;
-//     return 42;
-//   }
+  assert(findVarType(m, rr, "i") == dType->idxType());
```

Comment:
We have a helper called `resolveTypesOfVariables`. Would that help you here?

Reply:
Maybe so, I had just left the tests as they were without thinking about refactoring.

But I don't think I can use `resolveTypesOfVariables` for everything, because we also test for an AssociatedAction and with `resolveOnlyCandidate`.